src/python/pants/backend/python/tasks2/pex_build_util.py

```diff
@@ -119,6 +128,68 @@ def dump_requirements(builder, interpreter, req_libs, log, platforms=None):
       locations.add(dist.location)
 
 
+def build_python_distribution_from_target(target, workdir):
+  # Create a temporary working directory, pydistworkdir
+  pydist_workdir = os.path.join(workdir, 'pydistworkdir')
+  safe_mkdir(pydist_workdir)
+  pex_name = "%s.pex" % target.name
+  path_to_target = os.path.dirname(target.address.rel_path)
+
+  # disable cache to ensure reproducible runs
+  args = ['--disable-cache', path_to_target, '-o', os.path.join(pydist_workdir, pex_name)]
+  try:
+    pex_main.main(args=args)
+  except SystemExit as e:
+    raise TaskError(e)
+  except Exception as e:
+    raise TaskError(e)
+
+  # clean up egg-info created by setup.py
+  for path in glob.glob(os.path.join(path_to_target, '*.egg-info')):
+    shutil.rmtree(path)
+
+  # unzip into a chroot within the python dist workdir
+  zip_ref = zipfile.ZipFile(os.path.join(pydist_workdir, pex_name), 'r')
+  fingerprint = target.payload.fingerprint()
+  safe_mkdir(os.path.join(pydist_workdir, fingerprint))
+  zip_ref.extractall(os.path.join(pydist_workdir, fingerprint))
+  zip_ref.close()
+
+  # read the contents from .deps of the chroot to obtain the whl location
+  chroot_deps_contents = os.listdir(os.path.join(pydist_workdir, fingerprint, '.deps'))
+  if chroot_deps_contents:
+    whl_dist = chroot_deps_contents[0]  # TODO: find better way to grab .whl from chroot 
+    whl_location = os.path.join(pydist_workdir, fingerprint, '.deps', whl_dist)
+    return whl_location
+  return None
+
+
+def dump_python_distributions(builder, dist_targets, workdir, log):
+  """Dump python distribution targets into a given builder
+
+  :param builder: Dump the python_distributions into this builder.
+  :param dist_targets: PythonDistribution targets to dump
+  :param workdir: Working directory for python targets (./pantsd/python)
+  :param log: Use this logger.
+  """
+
+  # build whl for target using pex wheel installer
+  locations = set()
+  for tgt in dist_targets:
+    whl_location = build_python_distribution_from_target(tgt, workdir)
+    if whl_location:
+      locations.add(whl_location)
```

Comment:
if two different `python_dist` targets here key to the same wheel, that should probably be an error because otherwise one will be silently ignored.

Reply:
:white_check_mark: